### PR TITLE
Bugfix/issue 2291 explode style refactor

### PR DESCRIPTION
**Pull request by ralphdoe**
The explode and style now have default values depending on the parameter type
Issue: #2291

### CODE REVIEW

File: modules/swagger-core/src/main/java/io/swagger/util/ParameterProcessor.java [hunk 1]
```diff
@@ -144,6 +149,44 @@ public static Parameter applyAnnotations(OpenAPI openAPI, Parameter parameter, T
...
+                if (!schema.type().equals("object") && !schema.type().equals("array")) {
+                    explode = false;
+                }
+            }
+        }
+        return explode;
+    }
+
+    public static void setParameterStyle(Parameter parameter, io.swagger.oas.annotations.Parameter p) {
+        if (isExplodable(p)) {
```
Comment (webron) [inline modules/swagger-core/src/main/java/io/swagger/util/ParameterProcessor.java]: Parameter style doesn't depend on it being explodable or not.

File: modules/swagger-core/src/main/java/io/swagger/util/ParameterProcessor.java [hunk 2]
```diff
@@ -144,6 +149,44 @@ public static Parameter applyAnnotations(OpenAPI openAPI, Parameter parameter, T
...
+            if (StringUtils.isNotBlank(p.style())) {
+                parameter.setStyle(Parameter.StyleEnum.valueOf(p.style()));
+            } else {
+                if (HEADER.equals(p.in()) || PATH.equals(p.in())) {
```
Comment (webron) [inline modules/swagger-core/src/main/java/io/swagger/util/ParameterProcessor.java]: Just like with `explode`, if no value is provided by the user, we can assume the default value, and in that case it doesn't need to be serialized.

File: modules/swagger-core/src/main/java/io/swagger/util/ParameterProcessor.java [hunk 3]
```diff
@@ -144,6 +149,44 @@ public static Parameter applyAnnotations(OpenAPI openAPI, Parameter parameter, T
...
+                }
+            }
+        }
+        return explode;
+    }
+
+    public static void setParameterStyle(Parameter parameter, io.swagger.oas.annotations.Parameter p) {
+        if (isExplodable(p)) {
+            if (StringUtils.isNotBlank(p.style())) {
+                parameter.setStyle(Parameter.StyleEnum.valueOf(p.style()));
```
Comment (webron) [inline modules/swagger-core/src/main/java/io/swagger/util/ParameterProcessor.java]: The `style` value needs to be in lowercase - based on the tests, it is not.

File: modules/swagger-core/src/main/java/io/swagger/util/ParameterProcessor.java (hunk 2 above)
Comment (ralphdoe) [inline modules/swagger-core/src/main/java/io/swagger/util/ParameterProcessor.java]: Done!

File: modules/swagger-core/src/main/java/io/swagger/util/ParameterProcessor.java (hunk 1 above)
Comment (ralphdoe) [inline modules/swagger-core/src/main/java/io/swagger/util/ParameterProcessor.java]: Done!

File: modules/swagger-core/src/main/java/io/swagger/util/ParameterProcessor.java (hunk 3 above)
Comment (ralphdoe) [inline modules/swagger-core/src/main/java/io/swagger/util/ParameterProcessor.java]: Done! The Enums are now in lower case: mapper.configure(SerializationFeature.WRITE_ENUMS_USING_TO_STRING, true);

File: modules/swagger-jaxrs2/src/main/java/io/swagger/jaxrs2/OperationParser.java [hunk 4]
```diff
@@ -83,7 +85,7 @@
             parameterObject.setRequired(parameter.required());
             isEmpty = false;
         }
-        parameterObject.setStyle(StringUtils.isNoneBlank(parameter.style()) ? Parameter.StyleEnum.valueOf(parameter.style()) : null);
+        parameterObject.setStyle(StringUtils.isNoneBlank(parameter.style()) ? Parameter.StyleEnum.valueOf(parameter.style().toUpperCase()) : null);
```
Comment (ralphdoe) [inline modules/swagger-jaxrs2/src/main/java/io/swagger/jaxrs2/OperationParser.java]: Sure, there's a method that handles the style values. This line was deleted.

**Pull request merged**